lib/nats.js

```diff
@@ -824,10 +843,13 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   } else {
     psub = 'PUB ' + subject + SPC + opt_reply + SPC;
   }
+
   this.sendCommand(psub + Buffer.byteLength(msg) + CR_LF + msg + CR_LF);
 
   if (opt_callback !== undefined) {
     this.flush(opt_callback);
+  } else if (this.closed) {
+    this.emit('error', PUB_CONN_CLOSED_ERR);
```

Comment:
In Node, that _is_ the pattern.  Your error handlers and your process control _are_ your callbacks.  Node is extremely callback-oriented, and it is designed to efficiently toss arbitrary functions onto the event stack.  That's how it works.